pkg/v1/tkg/tkgctl/utils.go

```diff
@@ -0,0 +1,17 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package tkgctl
+
+import "github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/constants"
+
+func setCEIPOptinBasedOnConfigAndBuildEdition(t *tkgctl, edition string) string {
+	ceipOptIn, err := t.TKGConfigReaderWriter().Get(constants.ConfigVariableEnableCEIPParticipation)
```

Comment:
Do we really need to pass `tkgctl` client as function argument? Couple of options I can suggest are: 
1. It would be better if we can take just the `t.TKGConfigReaderWriter()` as function parameter instead of `tkgctl` client to make function more generic.
2. Or we can make this function part of `tkgctl` client definition so that we do not need to pass `tkgctl` client as function argument and invoke directly with `t,setCEIPOptinBasedOnConfigAndBuildEdition(edition)`